ibm/service/kubernetes/resource_ibm_container_vpc_worker.go

```diff
@@ -0,0 +1,577 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package kubernetes
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/bluemix-go/bmxerror"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+// Mutex to make resource creation sequential.
+var resourceIBMContainerVpcWorkerCreateMutex sync.Mutex
+var commonVarMutex sync.Mutex
+
+// Status of worker replace
+var workerReplaceStatus bool = false
+var replaceInProgress bool = false
+
+// Variable to identify the first run
+var initRun int = 1
+
+func ResourceIBMContainerVpcWorker() *schema.Resource {
+
+	return &schema.Resource{
+		Create:   resourceIBMContainerVpcWorkerCreate,
+		Read:     resourceIBMContainerVpcWorkerRead,
+		Delete:   resourceIBMContainerVpcWorkerDelete,
+		Exists:   resourceIBMContainerVpcWorkerExists,
+		Importer: &schema.ResourceImporter{},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
```

Comment:
To be inline with other resources ca you change to cluster (it shd be able to add take name or id)

Reply:
updated to "cluster_name"